gobblin-utility/src/main/java/gobblin/util/HadoopUtils.java

```diff
@@ -656,9 +660,37 @@ public static void safeRenameRecursively(FileSystem fileSystem, Path from, Path
   }
 
   public static Configuration getConfFromState(State state) {
+    return getConfFromState(state, Optional.<String> absent());
+  }
+
+  /**
+   * Provides Hadoop configuration given state.
+   * It also supports decrypting values on "encryptedPath".
+   * Note that this encryptedPath path will be removed from full path of each config key and leaving only child path on the key(s).
+   * If there's same config path as child path, the one stripped will have higher priority.
+   *
+   * e.g:
+   * - encryptedPath: writer.fs.encrypted
+   *   before: writer.fs.encrypted.secret
+   *   after: secret
+   *
+   * Common use case for these encryptedPath:
+   *   When there's have encrypted credential in job property but you'd like Filesystem to get decrypted value.
+   *
+   * @param srcConfig source config.
+   * @param encryptedPath Optional. If provided, config that is on this path will be decrypted. @see ConfigUtils.resolveEncrypted
+   *                      Note that config on encryptedPath will be included in the end result even it's not part of includeOnlyPath
+   * @return Hadoop Configuration.
+   */
+  public static Configuration getConfFromState(State state, Optional<String> encryptedPath) {
```

Comment:
You mean to have static path for Hadoop Config instead of getting it from caller?
The reason I've delegated this prefix out to the caller instead of having a static path in HadoopUtils is for the case of naming collision.
For example, if we need different credential for source Filesystem and target Filesystem, it would be hard to disambiguate using static path.
e.g:
File system accepts property "credential"
prefix for Source: src.encrypted.conf
prefix for Writer: writer.encrypted.conf